src/caf/core/zoning.py

```diff
@@ -725,6 +749,231 @@ class ZoningSystemMetaData(ctk.BaseConfig):
+        if segment_name not in self._segment_zoning:
+            return self._default_zoning
+        return self._segment_zoning[segment_name]
+
+    @property
+    def unique_zoning(self) -> dict[str, ZoningSystem]:
+        """Dict[str, ZoningSystem]: Dictionary containing a lookup of all
+        the unique `ZoningSystem` provided for the different segments.
+        The keys are the zone system name and values are the
+        `ZoningSystem` objects.
+        """
+        if self._unique_zoning is None:
+            self._unique_zoning = dict()
+            for zoning in self._segment_zoning.values():
+                if zoning.name not in self._unique_zoning:
+                    self._unique_zoning[zoning.name] = zoning
+            self._unique_zoning[self._default_zoning.name] = self._default_zoning
+        return self._unique_zoning
+
+    def zoning_groups(self) -> tuple[ZoningSystem, list[str]]:
+        """Iterates through the unique zoning systems and provides list of segments.
+
+        Yields
+        ------
+        ZoningSystem
+            Zone system for this group of segments.
+        List[str]
+            List of segment names which use this zone system.
+        """
+        zone_name = lambda s: self.get_zoning(s).name
+        zone_ls = sorted(self._segmentation.names, key=zone_name)
+        for zone_name, segments in itertools.groupby(zone_ls, key=zone_name):
+            zoning = self.unique_zoning[zone_name]
+            yield zoning, list(segments)
+
+    def __iter__(self) -> tuple[ZoningSystem, list[str]]:
+        """See `BalancingZones.zoning_groups`."""
+        return self.zoning_groups()
+
+    class BalancingConfClass(ctk.BaseConfig):
+        seg_conf: SegmentationInput
+        zon_conf: ZoningSystemMetaData
+        seg_zon: dict[str, ZoningSystemMetaData]
+
+    def save(self, path: Path) -> None:
+        """Saves balancing zones to output file.
+
+        Output file is saved to a yaml file
+
+        Parameters
+        ----------
+        path : Path
+            Path to output file to save.
+        """
+        seg_zon_dict = dict()
+        for name, zoning in self._segment_zoning.items():
+            seg_zon_dict[name] = zoning.metadata
+
+        out_conf = self.BalancingConfClass(
+            seg_conf=self._segmentation.input,
+            zon_conf=self._default_zoning.metadata,
+            seg_zon=seg_zon_dict,
+        )
+        out_conf.save_yaml(path)
+
+    @classmethod
+    def load(cls, path: Path) -> BalancingZones:
+        """Load balancing zones from config file.
+
+        Parameters
+        ----------
+        path : Path
+            Path to config file, should be the format defined
+            by `configparser` with section names defined in
+            `BalancingZones.OUTPUT_FILE_SECTIONS`.
+
+        Returns
+        -------
+        BalancingZones
+            Balancing zones with loaded parameters.
+        """
+        conf = cls.BalancingConfClass.load(path)
```

Comment:
The method is `load_yaml` not `load`
```suggestion
        conf = cls.BalancingConfClass.load_yaml(path)
```